File: ares-forge/src/main/java/dev/tigr/ares/forge/impl/modules/movement/SafeWalk.java
package dev.tigr.ares.forge.impl.modules.movement;

import dev.tigr.ares.core.feature.module.Category;
import dev.tigr.ares.core.feature.module.Module;
import dev.tigr.ares.forge.event.events.movement.WalkOffLedgeEvent;
import dev.tigr.simpleevents.listener.EventHandler;
import dev.tigr.simpleevents.listener.EventListener;
import net.minecraft.entity.MoverType;
import net.minecraft.entity.player.EntityPlayer;

/**
 * @author Tigermouthbear
 */
@Module.Info(name = "SafeWalk", description = "Keeps you from walking off ledges", category = Category.MOVEMENT)
public class SafeWalk extends Module {
    @EventHandler
    public EventListener<WalkOffLedgeEvent> walkOffLedgeEvent = new EventListener<>(event -> event.setCancelled(true));
}

File: src/main/kotlin/org/aresclient/ares/impl/instrument/module/modules/movement/SafeWalk.kt
package org.aresclient.ares.impl.instrument.module.modules.movement

import org.aresclient.ares.api.instruments.Module

object SafeWalk: Module(Category.MOVEMENT, "SafeWalk", "Keeps you from walking off ledges") {
    private val tolerance = settings.addDouble("Tolerance", 0.0).setMin(0.0).setMax(0.5)

    fun getTolerance(): Double = tolerance.value

    // see MixinPlayerEntity
}
